packages/rebound-router/lib/rebound-router.js

```diff
@@ -19,6 +19,19 @@ var SUCCESS = 'success';
 var ERROR = 'error';
 var LOADING = 'loading';
 
+// If the string passed to this function is a valid query string, split its values
+// out into a hash and return the object.
+function parseQueryString(str){
+  var res = {};
+  if(!str || str.indexOf('=') == -1) return str;
+  str = str.split('&');
+  str.forEach(function(item){
+    item = item.split('=');
+    res[item[0]] = item[1];
+  });
+  return res;
+};
```

Comment:
I'm having some trouble with this API design, @jmeas, @matthewrobb, @melissaroman,  let me know if this makes sense to you.

Consider the following routable component:

```html
<element name="test-page">
  <template>
    <h1>Test Page!</h1>
  </template>
  <script>
    return ({
      routes: {
        'test?:query': 'routeCallback'
      },
      routeCallback: function(queryParams){
        if(queryParams.foo){
            this.set('foo', queryParams.foo.substr(0,3));
        }
      }
    })
  </script>
</element>
```
With the proposed API above, If a user routes to `/test?foo=1234`, then:
```javascript
queryParams = { 
   foo: '1234'
};
```
The callback will successfully set `foo` to the first three characters of the query param.

But, if a user routes to `/test?foo=1234&foo=5678`, then:
```javascript
queryParams = { 
   foo: ['1234', '5678']
};
```
And the callback will fail saying that an object of type Array has no method substr... We need a consistent structure for parsed query params so users can make assumptions about how they can interact with it without type checking if the value is an array or not.

So, lets say we always return an array that contains the parsed query params, even if there is only one value. So, if a user routes to `/test?foo=1234`, then:
```javascript
queryParams = { 
   foo: ['1234']
};
```
And the route callback would now be: 
``` javascript
routeCallback: function(queryParams){
  if(queryParams.foo){
      this.set('foo', queryParams.foo[0].substr(0,3));
  }
}
```

Also, for the sake of consistency, query params without a value should be set to empty strings. So, if a user routes to `/test?foo`, then:
```javascript
queryParams = { 
   foo: ['']
};
```
In addition to not breaking string operations like the one above, this also means that checks for "truthyness" will pass as expected (ex: `if(queryParams.foo)`) whereas before, if the value was not wrapped in an array, checks for truthiness would not pass for query params with no value (ex: `queryParams = { foo: '' }` or `queryParams = {foo: undefined}` would both be falsy.

Consumers can now make assumptions about the structure of the query params passed to their callbacks and not worry about type checking to make sure their routes won't error out in odd corner cases. But, we loose some convenience because we can no longer write callbacks like the one below if we want to set all of our query params on our component – because they will all be wrapped in an array (probably not what people want in the majority of use cases). Users will have to manually pick properties from the arrays and set them on the component: 
``` javascript
// queryParams = { foo: ['bar'], biz: ['baz']}
routeCallback: function(queryParams){
  this.set(queryParams)
}
```
would have to become:
``` javascript
// queryParams = { foo: ['bar'], biz: ['baz']}
routeCallback: function(queryParams){
  this.set('foo', queryParams.foo[0]);
  this.set('biz', queryParams.biz[0])
}
```

Any thoughts? Is the lack of convenience of being able to set query params directly onto the component acceptable for less error-prone code?